packages/core/src/utils/oca.ts

```diff
@@ -11,32 +11,73 @@ import {
 import { W3cCredentialDisplay } from '../modules/openid/types'
 import { BrandingOverlay } from '@bifold/oca'
 
-export const buildFieldsFromAnonCredsCredential = (credential: CredentialExchangeRecord): Array<Field> => {
-  return credential?.credentialAttributes?.map((attr) => new Attribute(attr)) || []
+export type FiledExt = {
+  field: Attribute
+  attribute_name: string
 }
 
-export const buildFieldsFromW3cCredsCredential = (value: W3cCredentialDisplay): Array<Field> => {
-  return (
-    Object.entries(value.attributes)
-      .filter(([key]) => key !== 'id')
-      .map(([key, value]) => {
-        let formattedValue: string | number | null
+export const getAttributeField = (display: W3cCredentialDisplay, searchKey: string): FiledExt | undefined => {
+  let attributeName: string = 'Unknown'
+  let attributeValue: string | number | null = 'Unknown'
 
-        if (typeof value === 'object' && value !== null) {
-          formattedValue = JSON.stringify(value) // Convert object to string
-        } else {
-          formattedValue = value as string | number | null
-        }
+  for (const [key, value] of Object.entries(display.attributes)) {
+    let formattedValue: string | number | null
 
-        return new Attribute({
-          name: key,
-          value: formattedValue,
-          mimeType: typeof value === 'number' ? 'text/number' : 'text/plain',
-        })
-      }) || []
+    if (searchKey === key) {
+      if (typeof value === 'object' && value !== null) {
+        formattedValue = JSON.stringify(value) // Convert object to string
+      } else {
+        formattedValue = value as string | number | null
+      }
+
+      attributeName = key
+      attributeValue = formattedValue
+    }
+  }
+
+  //Now check credentialSubject for attributeName mapping
+  const credentialSubject = display.credentialSubject
+
+  if (credentialSubject) {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    for (const [key, value] of Object.entries(credentialSubject)) {
+      if (key !== searchKey) continue
+      if (!value) continue
```

Comment:
You're using both variables so you can safely remove the eslint exception, no? Can also merge those two conditionals
```suggestion
    for (const [key, value] of Object.entries(credentialSubject)) {
      if (key !== searchKey || !value) continue
```